proj/funcDictionary.py

```diff
@@ -0,0 +1,335 @@
+
+    # info: make empty list of nets:
+    nets = list()
+    for i in range(len(lsHist[0])+1):
+        nets.append(list())
+
+    # info: insert every sequence in the net for its size.
+    #     Thus all sequences with one size are in one net
+    for seqEl in seq:
+        seqElIndex = int((len(seqEl) - lsHist[1][0])/3)
+        nets[seqElIndex].append(seqEl)
+    return nets
+def loadSequence(step, plotData, i = -1, isExtractNum = True):
+    """
+    info: - load the nucleic acid sequences, generated from SONIA 
+        (and stored in a file 'pre.txt' or 'post.txt')
+          - cut of all acids after a certain length, defined by 
+              the threshold "extractNum"
+          - store the sequences in the list "seq" and the lengths of the 
+        the sequences in "ls"
+    input: step: 0 -> data before the Thymus selection (load from pre.txt)
+                 1 -> data after the Thymus selection (load from post.txt)
+        plotData: PlotData object, that contains the important data and parameters;
+            for more detail see documentation in plotData.py
+        i: int number of letters to be extracted from each strain, e.g. if i = 18,
+            then only the first 18 letters from each strain are extracted, the rest
+            is ignored
+        isExtractedNum: Bool, that says, if only a limited number of 
+            letters should be extracted; Thus, if isExtractNum == False, it makes no
+            sense, to set i > -1, because everything is extracted anyway.
+            If isExtractNum == True, i must be set to a certain value > -1.
+    output: a: list, containing all the loaded data
+        a4: list, containing the nucleic acid sequences
+        seq: list, containing the nucleic acid sequences first acids until 
+            the upper threshold "extractNum"
+        filename: string (either 'pre.txt' or 'post.txt')
+        ls: list of length values for each sequence in seq
+    """
+    # info: define filename
+    if step == 0:
+        filename = 'pre.txt'
+    if step == 1:
+        filename = 'post.txt'
+
+    # info: check, if plotData has an Ns-array, then choose N = plotData.Ns[i]; alternatively choose N = plotData.N[i]
+    if i > -1:
+        try: 
+            N = plotData.Ns[i]
+        except: 
+            print("\n Could not load element plotData.Ns[i]. Perhaps it does not exist?")
+            N = -1
+    else:
+        N = plotData.N
+
+    # info: make variables and lists as empty or zero objects
+    index = 0
+    a = list()
+    a4 = list()
+    seq = list()
+    ls = list() # right place to define ls?
+
+    # info: opening the list of generated sequences 
+    # info: (sequences generated done with SONIA, 
+    # info: which simulates VDJ recombination)
+    
+    with open(filename) as f:
+        for line in f.readlines():
+            if index < N:
+                a.append(line.split('\t'))
+                # info: attach the fourth element  to a4
```

Comment:
What is the fourth element? It's not clear why you use the fourth element here.

Reply:
See exlanation in the text

Reply:
Do you mean strings instead of strains below? The 4th column contains the strings, which are relevant ... ? Then I think that I understand what you mean here.